assets/js/phoenix_live_view/live_socket.js

```diff
@@ -519,7 +526,7 @@ export default class LiveSocket {
 
     this.boundTopLevelEvents = true
     // enter failsafe reload if server has gone away intentionally, such as "disconnect" broadcast
-    this.socket.onClose(event => {
+    this.serverCloseRef = this.socket.onClose(event => {
```

Comment:
As we only run `bindTopLevelEvents` once (we set `this.boundTopLevelEvents = true` above and do an early return if set), a user manually calling `liveSocket.disconnect()` and then `liveSocket.connect()` would not re-attach the onClose listener. Not sure if this is a real problem though, as this doesn't usually happen. We could move the onClose attach call to `connect`.